### PR TITLE
Feature/is over

### PR DESCRIPTION
**Pull request by XavierFavier**

### CODE REVIEW

File: game_board/src/Main.java [hunk 1]
```diff
@@ -1,5 +1,10 @@
+import fr.lecampusnumerique.cda2025.javaalgo.boardgames.PlayerTurns.PlayerTurn;
+import fr.lecampusnumerique.cda2025.javaalgo.boardgames.PlayerTurns.PlayerTurns;
+
 public class Main {
     public static void main(String[] args) {
         System.out.println("Hello world!");
+        PlayerTurns playerTurns = new PlayerTurn();
+        playerTurns.getPosition();
```
Comment (AnaisBonin) [inline game_board/src/Main.java]: If I well remember, we were advised to create a separate class MainDebug to test our code and leave Main alone :)

**Review by boubacargadio — APPROVED** [review]

File: game_board/src/fr/lecampusnumerique/cda2025/javaalgo/boardgames/PlayerTurns/PlayerTurn.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                getPosition = Integer.parseInt(scanner.next());
+            } catch (Exception e) {
+                System.out.println("Entry a number!");
+                continue;
+            }
+            //            if (getPosition <= 0 && getPosition > sizeGame) {
+            //                System.out.println("Position: " + getPosition);
+            //                break;
+            //            }
+
```
Comment (AnaisBonin) [inline game_board/src/fr/lecampusnumerique/cda2025/javaalgo/boardgames/PlayerTurns/PlayerTurn.java:77]: Do we need to keep these comments ? Are there here waiting for another feature to be developed?

File: game_board/src/fr/lecampusnumerique/cda2025/javaalgo/boardgames/PlayerTurns/PlayerTurns.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package fr.lecampusnumerique.cda2025.javaalgo.boardgames.PlayerTurns;
+
+import fr.lecampusnumerique.cda2025.javaalgo.boardgames.players.Player;
+
+import java.util.Scanner;
+
+public interface PlayerTurns {
+    Player switchPlayer();
+
+    boolean checkMove();
```
Comment (AnaisBonin) [inline game_board/src/fr/lecampusnumerique/cda2025/javaalgo/boardgames/PlayerTurns/PlayerTurns.java:10]: in order to fit the way we name our boolean methods and attributs, how could we rename this one?

getIsMoveCompatible ?

File: game_board/src/fr/lecampusnumerique/cda2025/javaalgo/boardgames/games/GameClass.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
+package fr.lecampusnumerique.cda2025.javaalgo.boardgames.games;
+
+public class GameClass implements Game {
+    //XF.
```
Comment (AnaisBonin) [inline game_board/src/fr/lecampusnumerique/cda2025/javaalgo/boardgames/games/GameClass.java]: What is this comment for? :)

File: game_board/src/fr/lecampusnumerique/cda2025/javaalgo/boardgames/games/GameClass.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                        if(myTable[a-c][b+c].getRepresentation() != curStr) {
+                            same = false;
+                        }
+                    }
+                    if(isOver_util(same, curStr)) {return true;}
+                }
+            }
+        }
+
+        return allCellFilled();
```
Comment (AnaisBonin) [inline game_board/src/fr/lecampusnumerique/cda2025/javaalgo/boardgames/games/GameClass.java]: Methods should only do one thing: we should split this code in several methods that have one responsibility: 
- checkDiagonal1
- checkDiagonal2
- checkColomn...

And is Over is a methods that call all of them one after the other- and its responsibility is to congragate all the different step of this check :)

**Review by AnaisBonin — CHANGES_REQUESTED** [review]
Goood! I left comments for this PR on things that I think should be addressed :)

**Comment by XavierFavier** [comment]
Thanks for your suggestions Anais! I will implement them.

**Comment by XavierFavier** [comment]
I updated it. Tell me what you think Anais 🦊

**Comment by AnaisBonin** [comment]
Let's merge this as is - but keep in mind two comments:
- rename CheckMove in Player Turn
- fix commented lines

File: game_board/src/fr/lecampusnumerique/cda2025/javaalgo/boardgames/PlayerTurns/PlayerTurn.java (hunk 2 above)
Comment (AnaisBonin) [reply game_board/src/fr/lecampusnumerique/cda2025/javaalgo/boardgames/PlayerTurns/PlayerTurn.java:77]: Keep in mind this comment for next feature branch review

**Review by AnaisBonin — APPROVED** [review]

**Pull request merged**